subprojects/gstreamer/libs/gst/base/gstaggregator.c

```diff
@@ -3239,6 +3242,18 @@ gst_aggregator_pad_chain_internal (GstAggregator * self,
 
       GST_DEBUG_OBJECT (self, "Selecting start time %" GST_TIME_FORMAT,
           GST_TIME_ARGS (start_time));
+
+      if (self->priv->start_time_selection ==
+          GST_AGGREGATOR_START_TIME_SELECTION_FIRST && !self->priv->running) {
+        PAD_UNLOCK (aggpad);
+        GST_OBJECT_UNLOCK (self);
+
+        gst_aggregator_start_srcpad_task (self);
```

Comment:
I think we need to add the same check to the gst_aggregator_src_pad_activate_mode_func, around
```
      case GST_PAD_MODE_PUSH:
      {
        GST_INFO_OBJECT (pad, "Activating pad!");
        gst_aggregator_start_srcpad_task (self);
        return TRUE;
      }
```

so it will become
```
      case GST_PAD_MODE_PUSH:
      {
        GST_INFO_OBJECT (pad, "Activating pad!");
        if (self->priv->start_time_selection != GST_AGGREGATOR_START_TIME_SELECTION_FIRST)
           gst_aggregator_start_srcpad_task (self);
        return TRUE;
      }
```

Reply:
okay I will test these changes and update the PR.

Reply:
@smomin7  properly document these changes in PR for later reference.